code/game/machinery/camera/trackable.dm

```diff
@@ -82,47 +61,140 @@
 	var/list/targets = sort_list(humans) + sort_list(others)
 	return targets
 
-///Toggles whether or not we're tracking something. Arg is whether it's on or off.
-/datum/trackable/proc/set_tracking(on = FALSE)
-	if(on)
+/// Takes a mob to track, resets our state and begins trying to follow it
+/// Best we can at least
+/datum/trackable/proc/set_tracked_mob(mob/living/track)
+	set_rechecking(FALSE)
+	if(tracked_mob)
+		UnregisterSignal(tracked_mob, list(COMSIG_QDELETING, COMSIG_MOVABLE_MOVED, COMSIG_MOVABLE_UPDATE_GLIDE_SIZE))
+	if(track && !isliving(track))
+		tracked_mob = null
+		return
+	tracked_mob = track
+	if(tracked_mob)
+		RegisterSignal(tracked_mob, COMSIG_QDELETING, PROC_REF(target_deleted))
+		RegisterSignal(tracked_mob, COMSIG_MOVABLE_MOVED, PROC_REF(target_moved))
+		RegisterSignal(tracked_mob, COMSIG_MOVABLE_UPDATE_GLIDE_SIZE, PROC_REF(glide_size_changed))
+		attempt_track()
+
+/datum/trackable/proc/target_deleted(datum/source)
+	SIGNAL_HANDLER
+	reset_tracking()
+
+/datum/trackable/proc/perspective_reset(atom/source)
+	SIGNAL_HANDLER
+	reset_tracking()
+
+/datum/trackable/proc/target_moved(datum/source)
+	SIGNAL_HANDLER
+	if(attempt_track())
+		return
+	set_rechecking(TRUE)
+
+/// Controls if we're processing to recheck the conditions that prevent tracking or not
+/datum/trackable/proc/set_rechecking(should_check)
+	if(should_check)
```

Comment:
could you turn this into an early return as well

Reply:
if you want, tho it feels icky to me since there's no like, failure condition to me. may just be cause I wrote it idk

Reply:
I can see your point